### PR TITLE
Add `confirm_global_rename` option to ease rename confirmation prompt

### PR DESCRIPTION
**Pull request by jonpalmisc**
This PR resolves my recently-created issue: https://github.com/sublimelsp/LSP/issues/1922

This worked in my own testing and I believe I followed all the appropriate conventions. Please let me know if I missed anything.

### CODE REVIEW

File: sublime-package.json [hunk 1]
```diff
@@ -11,6 +11,11 @@
               "default": false,
               "markdownDescription": "Run the server's formatProvider (if supported) on a document before saving. This option is also supported in syntax-specific settings and/or in the `\"settings\"` section of project files."
             },
+            "confirm_global_rename": {
```
Comment (predragnikolic) [inline sublime-package.json:14]: This is not in the right place for the `"confirm_global_rename"`.

Currently it is put in JSONSchema definitions,
but you want to put it here https://github.com/sublimelsp/LSP/blob/60d725264e8fafe9111cb1ab8373f9829cc63f79/sublime-package.json#L288

**Review by predragnikolic — CHANGES_REQUESTED** [review]
Looks good, but requires a small change.

File: sublime-package.json (hunk 1 above)
Comment (jonpalmisc) [reply sublime-package.json]: This should be fixed now.

**Review by predragnikolic — APPROVED** [review]

**Comment by rchl** [comment]
I don't quite see how a dialog that can be confirmed with a single press of an enter key warrants a new option.

It's imo useful to know how many files are going to get changed, especially with many files open since those have to be found and saved manually.

Maybe you could describe what makes the dialog so annoying?

I'd like to figure out solution that is less or not annoying that can work for everybody rather than handling that with a new option.

For example maybe we could proceed with renaming without showing the dialog but then we'd show how many files were affected through a message in a panel. That could also act as a quick way of switching to each modified file.

**Comment by jonpalmisc** [comment]
I realize I could have provided more context in my original issue. Hopefully I can address some of these concerns.

> Maybe you could describe what makes the dialog so annoying?

Not including auto-complete, renaming is hands-down my most used feature. I easily use it 50+ times per day. In the ~2 months I've been using the LSP plugin, I have _never_ wanted to abort a rename operation. "Staying out of the way" is part of the spirit of Sublime Text, and I find this dialog — which interrupts my work on every rename operation — to not be aligned with this philosophy. 

> I don't quite see how a dialog that can be confirmed with a single press of an enter key warrants a new option.

This is my first contribution to this project, so I'm in unfamiliar territory here. I'll say this:
- The current behavior of the rename operation is unchanged with the addition of this option — I made the default value of the setting `true`, meaning "confirm rename operations".
- I imagine I am not the only one bothered by this behavior. I was actually surprised to find that this option didn't exist at first, as I thought "surely there is a way to disable this popup".

> It's imo useful to know how many files are going to get changed, especially with many files open since those have to be found and saved manually.

I disagree — I am never interested in how many files changed and I always hit "save all" afterwards. The changes will be clearly visible with a quick `git diff`. I imagine there are many users like you who do want to know, and many other users like me who don't want to know — as such, it makes sense for this to be a configurable option, meaning everyone can have their preferred experience.

> I'd like to figure out solution that is less or not annoying that can work for everybody rather than handling that with a new option.

Could you explain what's annoying about handling a new option? As I said before, this change will be transparent to existing plugin users as the default value of the option yields the current behavior. I don't see how adding this option complicates things or would have any negative effects.

> For example maybe we could proceed with renaming without showing the dialog but then we'd show how many files were affected through a message in a panel. That could also act as a quick way of switching to each modified file.

If this were the case, I'd still want the ability to disable it. A pop-up panel also sounds annoying.

**Comment by jwortmann** [comment]
I guess this confirm dialog is inspired by the built-in "Replace" functionality from the Find in Files panel, because it uses the exact same text message.

But our rename command will open a TextInputHandler in all cases, before the dialog prompt, right?

How about we use the following text in `preview()` for the preview area under the input field:
"<kbd>Enter</kbd> to confirm, <kbd>Ctrl</kbd>+<kbd>Enter</kbd> to preview the changes"

Then we drop the confirm dialog, and for the normal execution (Enter) we show a `window.status_message()` "Made 42 replacements in 5 files". If something was unexpected for the user, they can stil just undo it with <kbd>Ctrl</kbd>+<kbd>Z</kbd>.

And with <kbd>Ctrl</kbd>+<kbd>Enter</kbd> (or the corresponding key for macOS) we can still do the functionality from the "Dry Run" button. I think something similar to this with the modifier key was used in the initial version of the recent "Goto Diagnostic in quick panel" PR. I'm not exactly sure whether it also works with TextInputHandlers though.

**Comment by predragnikolic** [comment]
The confirm dialog was added to fix a case where someone renames a built-in type in pyls, like `str`. 
That caused more than 1000 files to be changed. 

The user couldn't not just press `ctrl+z` to undo that, because each file was edited with a different edit token. So the user must close each file manually and discard changes.

So the confirm modal was added to prevent cases like that.

Most editors do not have such a confirm modal, but rely on the textDocument/prepareRename which should confirm if a rename is valid for that range.

Most language servers wont even allow renaming of built in types. 

I find the confirmation modal useful. But I wouldn't mind if someone wants to disable the modal, given that LSP for ST implemented the textDocument/prepareRename.

**Comment by jwortmann** [comment]
> The user couldn't not just press `ctrl+z` to undo that, because each file was edited with a different edit token. So the user must close each file manually and discard changes.

Ah, you're right. I guess then we either need to wait for https://github.com/sublimehq/sublime_text/issues/4604 to be implemented in ST, or rely on the current confirmation dialog.

File: plugin/core/types.py [hunk 2]
```diff
@@ -198,6 +198,7 @@ class Settings:
     log_server = None  # type: List[str]
     lsp_code_actions_on_save = None  # type: Dict[str, bool]
     lsp_format_on_save = None  # type: bool
+    confirm_global_rename = None # type: bool
```
Comment (jwortmann) [inline plugin/core/types.py:201]: Seems that these entries are ordered alphabetically, and this one is inserted at a wrong place. The same applies to next next changed line below.

**Comment by jonpalmisc** [comment]
> The confirm dialog was added to fix a case where someone renames a built-in type in pyls, like str. That caused more than 1000 files to be changed.

Perhaps the confirmation dialog could simply be made less eager? Rather than alerting on >1 files modified, only alert on >10? I can see the benefit in that. I think that >1 files just seems like a low threshold as that is easily triggered when renaming a C++ class that is defined in a header and implemented in a source file.

**Comment by rwols** [comment]
> Perhaps the confirmation dialog could simply be made less eager? Rather than alerting on \>1 files modified, only alert on \>10? \>15? (The limit is arbitrary, really.) I can see the benefit in that.

I think this is a good approach. Can you change the setting into something like `rename_dialog_threshold` with a default value of `1`?

**Comment by jonpalmisc** [comment]
For unrelated reasons, I'm not going to be using Sublime Text as my editor moving forward, and will not be continuing work on this PR — this can be closed if nobody else wants to take up the work.

**Comment by predragnikolic** [comment]
@jonpalmisc thanks for taking the time to create the PR!

But I would close this PR tomorrow, if no one has nothing against.
We can revisit this, if other people also report that the rename popup is distracting.

**Pull request closed without merging**